File: template-dashboard/src/app/domain/reports/report-param.ts
import { Archive } from '../archive/archive';
import { ReportParamOption } from './report-param-option';

export class ReportParam {

    key: string;

    label: string;

    options?: ReportParamOption[];

    order: Number;

    required: Boolean;

    type: string;

    value?: any;
}

File: template-app/core/src/main/java/org/myorganization/template/reports/domain/reportparam/ReportParam.java
package org.myorganization.template.reports.domain.reportparam;

import java.io.Serializable;
import java.util.List;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter @Setter @NoArgsConstructor
public class ReportParam implements Serializable {

	private static final long serialVersionUID = -6003711279314865842L;

	private String type;
	
	private String key;
	
	private String value;
	
	private String label;
	
	private List<ReportParamOption> options;
	
	private Integer order;
	
	private Boolean required = false;

	private Boolean forPrompting  = true;
	
}
